notification-server/src/bin/service.rs

```diff
@@ -112,35 +112,59 @@ async fn traverse_chain(
                 continue;
             }
         };
-        for result in process(transactions).await.iter() {
+        for result in process(transactions).await.into_iter() {
             info!(
                 "Sending notification to account {} with type {:?}",
-                result.address, result.transaction_type
+                result.address(),
+                result.transaction_type()
             );
-
             let devices = loop {
                 match database_connection
                     .prepared
-                    .get_devices_from_account(result.address)
+                    .get_devices_from_account(result.address())
                     .await
                 {
                     Ok(devices) => break devices,
                     Err(err) => {
                         error!(
                             "Error retrieving devices for account {}: {:?}. Retrying...",
-                            result.address, err
+                            result.address(),
+                            err
                         );
                         sleep(DATABASE_RETRY_DELAY).await;
                     }
                 }
             };
-
-            for device in devices
+            let devices: Vec<_> = devices
                 .iter()
-                .filter(|device| device.preferences.contains(&result.transaction_type))
-            {
+                .filter(|device| device.preferences.contains(result.transaction_type()))
+                .collect();
+            if devices.is_empty() {
+                info!(
+                    "No devices subscribed to account {} having preference {:?}",
+                    result.address(),
+                    result.transaction_type()
+                );
+                continue;
+            }
+            let enriched_notification_information =
+                match result.enrich(concordium_client.clone(), block_hash).await {
+                    Ok(information) => information,
+                    Err(err) => {
+                        error!(
```

Comment:
As talked about we could also send what ever we have to the clients in this case.

It was mentioned by @Radiokot that they would prefer just to leave it out